tree/dataframe/test/dataframe_datasetspec.cxx

```diff
@@ -0,0 +1,449 @@
+#include <gtest/gtest.h>
+#include <ROOT/RDataFrame.hxx>
+#include <ROOT/RVec.hxx>
+#include <ROOT/RDFHelpers.hxx>
+#include <ROOT/TestSupport.hxx>
+#include <TSystem.h>
+
+#include <thread> // std::thread::hardware_concurrency
+
+using namespace ROOT;
+using namespace ROOT::RDF::Experimental;
+
+using namespace std::literals; // remove ambiguity of using std::vector<std::string>-s and std::string-s
+
+void EXPECT_VEC_EQ(const std::vector<ULong64_t> &vec1, const std::vector<ULong64_t> &vec2)
+{
+   ASSERT_EQ(vec1.size(), vec2.size());
+   for (auto i = 0u; i < vec1.size(); ++i)
+      EXPECT_EQ(vec1[i], vec2[i]);
+}
+
+// The helper class is also responsible for the creation and the deletion of the root specTestFiles
+// Reasons: each tests needs (almost) all files; the Snapshot action is not available in MT
+class RDatasetSpecTest : public ::testing::TestWithParam<bool> {
+protected:
+   RDatasetSpecTest() : NSLOTS(GetParam() ? std::min(4u, std::thread::hardware_concurrency()) : 1u)
```

Comment:
We should have a test where the input file actually has more than one cluster to properly exercise the logic in TTreeProcessorMT and MakeClusters

Reply:
I will add a TTreeProcessorMT specific test for that!